+  // Making cards on the swimlanes dragable 
+  componentDidMount(){
+    const containers = [
+      this.swimlanes.backlog.current,
+      this.swimlanes.inProgress.current,
+      this.swimlanes.complete.current
+
+    ]
+    Dragula(containers)
+  }
   getClients() {
     return [
       ['1','Stark, White and Abbott','Cloned Optimal Architecture', 'in-progress'],
-- 
2.42.0.windows.1


From ac80c97962089e60ee88531a4bad599433b1061c Mon Sep 17 00:00:00 2001
From: Sphesihle-Duma <dumasphesihle22@gmail.com>
Date: Tue, 30 Jul 2024 22:07:27 +0200
Subject: [PATCH 8/9] Updated Board.js and Swimlane to change the color of the
 cards

---
 src/Board.js    | 29 ++++++++++++++++++++++++++---
 src/Swimlane.js |  3 ++-
 2 files changed, 28 insertions(+), 4 deletions(-)

diff --git a/src/Board.js b/src/Board.js
index ff5f42c..da9a35a 100644
--- a/src/Board.js
+++ b/src/Board.js
@@ -28,8 +28,29 @@ export default class Board extends React.Component {
       this.swimlanes.inProgress.current,
       this.swimlanes.complete.current
 
-    ]
-    Dragula(containers)
+    ];
+    // changing the color of the cards when are drag and from and into the swimlanes
+
+    const drake = Dragula(containers)
+
+    drake.on('drag', (el) => {
+      el.classList.add('dragging')
+    })
+
+    drake.on('dragend', (el) => {
+      el.classList.remove('dragging')
+      el.classList.remove('Card-grey', 'Card-blue', 'Card-green')
+
+      if(el.closest('.backlog')){
+        el.classList.add('Card-grey')
+      }
+      else if(el.closest('.inprogress')){
+        el.classList.add('Card-blue')
+      }
+      else if(el.closest('.complete')){
+        el.classList.add('Card-green')
+      }
+    })
   }
   getClients() {
     return [
@@ -60,9 +81,11 @@ export default class Board extends React.Component {
       status: companyDetails[3],
     }));
   }
+  // Updating swimlane to use the class name props
   renderSwimlane(name, clients, ref) {
+    const swimlaneClass = name.replace(/\s+/g, '').toLowerCase()
     return (
-      <Swimlane name={name} clients={clients} dragulaRef={ref}/>
+      <Swimlane className={swimlaneClass} name={name} clients={clients} dragulaRef={ref}/>
     );
   }
 
diff --git a/src/Swimlane.js b/src/Swimlane.js
index 7e6e0d0..42654ff 100644
--- a/src/Swimlane.js
+++ b/src/Swimlane.js
@@ -16,7 +16,8 @@ export default class Swimlane extends React.Component {
       );
     })
     return (
-      <div className="Swimlane-column">
+      // Adding class same as the name of the swimlane
+      <div className={`Swimlane-column ${this.props.className}`}>
         <div className="Swimlane-title">{this.props.name}</div>
         <div className="Swimlane-dragColumn" ref={this.props.dragulaRef}>
           {cards}
-- 
2.42.0.windows.1


From 036c6592a5125363f376bbe8406ea8e905e17f94 Mon Sep 17 00:00:00 2001
From: Sphesihle-Duma <dumasphesihle22@gmail.com>
Date: Tue, 30 Jul 2024 22:10:20 +0200
Subject: [PATCH 9/9] changing the opacity of the dragged card

---
 src/Board.css | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Board.css b/src/Board.css
index 792e356..5a1d607 100644
--- a/src/Board.css
+++ b/src/Board.css
@@ -1,3 +1,8 @@
 .Board {
   margin-top: 12px;
 }
+
+
+.dragging {
+  opacity: 0.5;
+}
\ No newline at end of file
-- 
2.42.0.windows.1

